ark/internal/genai/memory_http.go

```diff
@@ -43,28 +43,73 @@ func NewHTTPMemory(ctx context.Context, k8sClient client.Client, memoryName, nam
 		return nil, fmt.Errorf("memory has no lastResolvedAddress in status")
 	}
 
-	sessionId := config.SessionId
-	if sessionId == "" {
-		sessionId = string(memory.UID)
-	}
-
 	// Create HTTP client with timeout for memory operations
 	httpClient := common.NewHTTPClientWithLogging(ctx)
 	if config.Timeout > 0 {
 		httpClient.Timeout = config.Timeout
 	}
 
+	baseURL := strings.TrimSuffix(*memory.Status.LastResolvedAddress, "/")
+
+	// Initialize conversation ID by calling cluster-memory
+	conversationId, err := initializeConversationID(ctx, httpClient, baseURL, config.ConversationId)
+	if err != nil {
+		return nil, fmt.Errorf("failed to initialize conversation ID: %w", err)
+	}
+
 	return &HTTPMemory{
 		client:           k8sClient,
 		httpClient:       httpClient,
-		baseURL:          strings.TrimSuffix(*memory.Status.LastResolvedAddress, "/"),
-		sessionId:        sessionId,
+		baseURL:          baseURL,
+		conversationId:   conversationId,
 		name:             memoryName,
 		namespace:        namespace,
 		eventingRecorder: memoryRecorder,
 	}, nil
 }
 
+// initializeConversationID calls cluster-memory to initialize or get a conversation ID
+func initializeConversationID(ctx context.Context, httpClient *http.Client, baseURL, conversationID string) (string, error) {
+	type initRequest struct {
+		ConversationID string `json:"conversation_id,omitempty"`
+	}
+
+	type initResponse struct {
+		ConversationID string `json:"conversation_id"`
+	}
+
+	reqBody, err := json.Marshal(initRequest{ConversationID: conversationID})
+	if err != nil {
+		return "", fmt.Errorf("failed to serialize request: %w", err)
+	}
+
+	requestURL := fmt.Sprintf("%s/conversations/init", baseURL)
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, requestURL, bytes.NewReader(reqBody))
+	if err != nil {
+		return "", fmt.Errorf("failed to create request: %w", err)
+	}
+
+	req.Header.Set("Content-Type", ContentTypeJSON)
+	req.Header.Set("User-Agent", UserAgent)
+
+	resp, err := httpClient.Do(req)
+	if err != nil {
+		return "", fmt.Errorf("HTTP request failed: %w", err)
+	}
+	defer func() { _ = resp.Body.Close() }()
```

Comment:
isn't this just enough? is there any pro wrapping it in a func?

```suggestion
	defer resp.Body.Close()
```